File: kotlin/src/main/kotlin/topics/strings.kt
package learn.kotlin.topics.strings
import java.util.*

val helloWorld = "Hello World"

fun main() {
    // String interpolation
    println("helloWorld: String = $helloWorld")

    ///String methods
    println("helloWorld.length: Int = ${helloWorld.length}")
    println("helloWorld.contains(\"World\"): Boolean = ${helloWorld.contains("World")}")
    println("helloWorld.startsWith(\"Hello\"): Boolean = ${helloWorld.startsWith("Hello")}")
    println("helloWorld.endsWith(\"World\"): Boolean = ${helloWorld.endsWith("World")}")
    println("helloWorld.indexOf(\"World\"): Int = ${helloWorld.indexOf("World")}")
    println("helloWorld.substring(6): String = ${helloWorld.substring(6)}")
    println("helloWorld.substring(6, 11): String = ${helloWorld.substring(6, 11)}")
    println("helloWorld.toUpperCase(): String = ${helloWorld.uppercase(Locale.getDefault())}")
    println("helloWorld.toLowerCase(): String = ${helloWorld.lowercase(Locale.getDefault())}")
    println("helloWorld.trim(): String = ${helloWorld.trim()}")
    println("helloWorld.split(\" \"): List<String> = ${helloWorld.split(" ")}")
    println("helloWorld.replace(\"World\", \"Kotlin\"): String = ${helloWorld.replace("World", "Kotlin")}")
    println("helloWorld.replaceFirst(\"World\", \"Kotlin\"): String = ${helloWorld.replaceFirst("World", "Kotlin")}")
    println("helloWorld.removeRange(0, 6): String = ${helloWorld.removeRange(0, 6)}")
    println("helloWorld.removePrefix(\"Hello \"): String = ${helloWorld.removePrefix("Hello ")}")
    println("helloWorld.removeSuffix(\" World\"): String = ${helloWorld.removeSuffix(" World")}")
    println("helloWorld.removeSurrounding(\"Hello \", \" World\"): String = ${helloWorld.removeSurrounding("Hello ", " World")}")
}

File: c#/Topics/Strings.cs
namespace Topics
{
  class Strings
  {
    static void stringExample(string[] args)
    {
      string s1 = "Hello";
      string s2 = "World";
      string s3 = s1 + " " + s2;
      Console.WriteLine(s3);
      Console.WriteLine(s3.Length);
      Console.WriteLine(s3.ToUpper());
      Console.WriteLine(s3.ToLower());
      Console.WriteLine(s3.Contains("Hello"));
      Console.WriteLine(s3.Contains("hello"));
      Console.WriteLine(s3.StartsWith("Hello"));
      Console.WriteLine(s3.StartsWith("hello"));
      Console.WriteLine(s3.EndsWith("World"));
      Console.WriteLine(s3.EndsWith("world"));
      Console.WriteLine(s3.IndexOf("World"));
      Console.WriteLine(s3.IndexOf("world"));
      Console.WriteLine(s3.LastIndexOf("World"));
      Console.WriteLine(s3.LastIndexOf("world"));
      Console.WriteLine(s3.Substring(6));
      Console.WriteLine(s3.Substring(6, 5));
      Console.WriteLine(s3.Replace("World", "C#"));
      Console.WriteLine(s3.Replace("world", "C#"));
      Console.WriteLine(s3.Insert(6, "C#"));
      Console.WriteLine(s3.Remove(6));
      Console.WriteLine(s3.Remove(6, 5));
      Console.WriteLine(s3.Trim());
      Console.WriteLine(s3.TrimStart());
      Console.WriteLine(s3.TrimEnd());
      Console.WriteLine(s3.PadLeft(20));
      Console.WriteLine(s3.PadRight(20));
      Console.WriteLine(s3.PadLeft(20, '*'));
      Console.WriteLine(s3.PadRight(20, '*'));
      Console.WriteLine(s3.PadLeft(20, ' '));
      Console.WriteLine(s3.PadRight(20, ' '));
      Console.WriteLine(s3.PadLeft(20, '0'));
      Console.WriteLine(s3.PadRight(20, '0'));
      Console.WriteLine(s3.PadLeft(20, '1'));
      Console.WriteLine(s3.PadRight(20, '1'));

      //interpolated string
      string s4 = $"{s1} {s2}";
      Console.WriteLine(s4);
    }
  }
}
